### ISSUE TITLE
Multiple dropdowns on same page

### ISSUE THREAD

**Issue by grumpykiwi** [Megabit/Blazorise#4017]
**Describe the bug**
I need to be able to have multiple select lists on a page, each with their own ID and value. The value comes from a List<T> object.

On submit of the page, not in an editform, I need to loop thru each row and detect the changed values for the select list/dropdown and write the change to a table. The ID of the select is set from the List<T> object and is unique on the page.

On page load each select/dropdown should have its value set and displayed. The dropdown value in each row should be changeable without affecting other rows.

How should I do this?  Which Blazorise component is most suited for this task?

Here is the relevant supporting code

```
@foreach (var e in Entries)
{
	<TableRow>
		<TableRowCell>@e.Dateval.DayOfWeek - @e.Dateval.ToShortDateString()</TableRowCell>
		<TableRowCell>@e.Clientusername</TableRowCell>
		<TableRowCell>@e.Clientcode</TableRowCell>
		<TableRowCell>@e.Timeval.ToString("N1")</TableRowCell>
		<TableRowCell>@e.Descval</TableRowCell>
		<TableRowCell>
			<DropdownList SelectedValueChanged="@OnSelectedValueChanged"
				TItem="BillingCode" 
				TValue="string"
				Data="@BillingCodes" id="@e.Uniqueid"
				TextField="@((item) => item.Description)"
				ValueField="@((item) => item.Value)">
				@(BillingCodes?.FirstOrDefault(x => x.Value == e.Billorff)?.Description ?? "Select item")
			</DropdownList>
		</TableRowCell>
	</TableRow>
}

public class BillingCode
{
	public int Id { get; set; }
    public string Value { get; set; }
	public string Description { get; set; }
	public int TCode { get; set; }
}

BillingCodes = await ctx.BillingCodes.OrderBy(e => e.Id).ToListAsync();
```

**Comment by David-Moreira** [comment]
Hello,

So is there something that's not working as expected or are you just looking for sugestions?
It seems like your example would work as long as you'd match each entry's Id and update it accordingly. Meaning that somehow you should track the state in each for each, maybe something like this, might help you?

An anonymous method that injects your contextual entry
`SelectedValueChanged="@( (value) => OnSelectedValueChanged(entry, value) )"`
I think something similar to this will work, is this what you're looking for?

**Comment by grumpykiwi** [comment]
I ended up going with EditForm and binding to a custom model so that I could keep track of the changed values in the dropdowns. I used InputSelect's and assigned each a unique value so I can loop thru them and update the DB accordingly.

Also moved my page setup code to OnInitializedAsync from OnAfterRenderAsync. That was causing strange behavior when I selected a different value in a dropdown. 

Thanks for your very prompt help.
